dune/grid/common/ZoltanPartition.cpp

```diff
@@ -122,11 +128,62 @@ std::vector<int> zoltanGraphPartitionGridOnRoot(const CpGrid& cpgrid,
         }
 #endif
     }
-    cc.broadcast(&parts[0], parts.size(), root);
+    // free space allocated for zoltan.
     Zoltan_LB_Free_Part(&exportGlobalGids, &exportLocalGids, &exportProcs, &exportToPart);
     Zoltan_LB_Free_Part(&importGlobalGids, &importLocalGids, &importProcs, &importToPart);
     Zoltan_Destroy(&zz);
-    return parts;
+
+    cc.broadcast(&parts[0], parts.size(), root);
+    std::vector<int> my_well_indices;
+
+    if( !pretendEmptyGrid)
```

Comment:
I assume your are talking about the if statement. `!pretendEmptyGrid` is only true on the root process that has the global grid during loadbalancing. I gues a more speaking name would be appreciated for readability? `hasGlobalGrid` or `isRootProcess`?
Anyway that process sends all the well information to the other processes.